source/state_representation/test/tests/space/cartesian/test_cartesian_pose.cpp

```diff
@@ -0,0 +1,254 @@
+  Eigen::Quaterniond rot_truth(1., 0., 0., 0.);
+  EXPECT_EQ(tf1.get_name(), "world");
+  EXPECT_EQ(tf1.get_reference_frame(), "t1");
+  expect_near(tf1.get_position(), pos_truth);
+  EXPECT_GT(abs(tf1.get_orientation().dot(rot_truth)), 1 - tol);
+}
+
+TEST_F(CartesianPoseTestClass, TestInverseNonNullOrientation) {
+  tf1.set_orientation(Eigen::Quaterniond(0.70710678, 0.70710678, 0., 0.));
+  tf1 = tf1.inverse();
+  Eigen::Vector3d pos_truth(-1, -3, 2);
+  Eigen::Quaterniond rot_truth(0.70710678, -0.70710678, 0., 0.);
+  expect_near(tf1.get_position(), pos_truth);
+  EXPECT_GT(abs(tf1.get_orientation().dot(rot_truth)), 1 - tol);
+}
+
+TEST_F(CartesianPoseTestClass, TestMultiplyInverseNonNullOrientation) {
+  tf1.set_orientation(Eigen::Quaterniond(0.70710678, 0.70710678, 0., 0.));
+  tf1 *= tf1.inverse();
+  Eigen::Vector3d pos_truth(0, 0, 0);
+  Eigen::Quaterniond rot_truth(1., 0., 0., 0.);
+  expect_near(tf1.get_position(), pos_truth);
+  EXPECT_GT(abs(tf1.get_orientation().dot(rot_truth)), 1 - tol);
+}
+
+TEST(CartesianPoseTest, MultiplyPoseAndState) {
+  CartesianPose p = CartesianPose::Random("test");
+  CartesianState s = CartesianPose::Random("test2", "test");
+  CartesianState res = p * s;
+  CartesianPose res2 = p * static_cast<CartesianPose>(s);
+  expect_near(res.get_position(), res2.get_position());
+  EXPECT_GT(abs(res.get_orientation().dot(res2.get_orientation())), 1 - 1e-5);
+}
+
+TEST_F(CartesianPoseTestClass, TestAddTwoPoses) {
+  tf1.set_position(Eigen::Vector3d::Zero());
+  Eigen::Vector3d pos2(1, 0, 0);
+  Eigen::Quaterniond rot2(0, 1, 0, 0);
+  tf2 = CartesianPose("t1", pos2, rot2);
+  auto tf3 = tf1 + tf2;
+  Eigen::Vector3d pos_truth(1, 0, 0);
+  Eigen::Quaterniond rot_truth(0, -1, 0, 0);
+  expect_near(tf3.get_position(), pos_truth);
+  EXPECT_GT(abs(tf3.get_orientation().dot(rot_truth)), 1 - 1e-5);
+}
+
+TEST_F(CartesianPoseTestClass, TestPoseToVelocity) {
+  tf1.set_orientation(Eigen::Quaterniond(0, 1, 0, 0));
+  std::chrono::seconds dt1(1);
+  std::chrono::milliseconds dt2(100);
+  auto res1 = tf1 / dt1;
+  EXPECT_EQ(res1.get_linear_velocity(), tf1.get_position());
+  EXPECT_EQ(res1.get_angular_velocity(), Eigen::Vector3d(M_PI, 0, 0));
+  auto res2 = tf1 / dt2;
+  EXPECT_EQ(res2.get_linear_velocity(), 10 * tf1.get_position());
+  EXPECT_EQ(res2.get_angular_velocity(), 10 * Eigen::Vector3d(M_PI, 0, 0));
+}
+
+TEST_F(CartesianPoseTestClass, TestImplicitConversion) {
+  CartesianTwist vel("t1");
+  vel.set_linear_velocity(Eigen::Vector3d(0.1, 0.1, 0.1));
+  vel.set_angular_velocity(Eigen::Vector3d(M_PI, 0, 0));
+  tf1 += vel;
+  Eigen::Vector3d pos_truth(1.1, 2.1, 3.1);
+  Eigen::Quaterniond rot_truth(0, 1, 0, 0);
+  EXPECT_EQ(tf1.get_position(), pos_truth);
+  EXPECT_GT(abs(tf1.get_orientation().dot(rot_truth)), 1 - 1e-5);
+}
+
+TEST(CartesianPoseTest, TestPoseDistance) {
+  CartesianPose p1("test",
+                   Eigen::Vector3d(-0.353849997774433, -0.823586525156853, -1.57705702279920),
```

Comment:
the formatting here with the new style guide should have brought this back to the continuation indent of 4 spaces, instead of aligning with the opening `(`. Not a big deal but I'm curious why it's like this.

Reply:
Because I didn't auto format the code it seems :smile: sorry